CHANGELOG.md

```diff
@@ -12,6 +12,120 @@
 - [go-mod-secrets](https://github.com/edgexfoundry/go-mod-secrets/blob/main/CHANGELOG.md)
 - [go-mod-configuration](https://github.com/edgexfoundry/go-mod-configuration/blob/main/CHANGELOG.md) (indirect dependency)
 
+## [4.0.0] Odessa - 2025-02-TBD (Only compatible with the 4.x releases)
+
+### Features✨  
+
+- Add message bus subscription log in core-data ([a6f6ea5…](https://github.com/edgexfoundry/edgex-go/commit/a6f6ea5060545076c40d55fda1b913bd4da7e70a))
+- Add query parameters for device parents/children([#5053](https://github.com/edgexfoundry/edgex-go/issues/5053)) ([a1c5a89…](https://github.com/edgexfoundry/edgex-go/commit/a1c5a899654eae7656c51fcf435c291ced400fbb))
+- Allow empty profileName in Device ([702ba25…](https://github.com/edgexfoundry/edgex-go/commit/702ba25b8e74ae87972de5b867e4c6885ac51f68))
+- Add regen token API in secretstore-setup ([#5079](https://github.com/edgexfoundry/edgex-go/issues/5079)) ([9ebed02…](https://github.com/edgexfoundry/edgex-go/commit/9ebed02dde4a816a76396a8e6ec635dd7fd91285))
+- Add Configurable Limits for Device and Resource Capacities ([#5077](https://github.com/edgexfoundry/edgex-go/issues/5077)) ([4896b45…](https://github.com/edgexfoundry/edgex-go/commit/4896b4575f29b11e72f5a58f0619c899671835f4))
+- Event retention enhancement ([c067ea6…](https://github.com/edgexfoundry/edgex-go/commit/c067ea6d19c3779c8d0085825964471478d87c5c))
+```text
+
+BREAKING CHANGE: Add a retention policy for auto events, enhancing time-based and count-based retention.
+
+```
+- Upgrade go-mod-messaging and remove event size check ([cafc722…](https://github.com/edgexfoundry/edgex-go/commit/cafc7220272db75275058deb04c0a161acd7b921))
+```text
+
+BREAKING CHANGE: Upgrade go-mod-messaging and remove event size check
+
+```
```

Comment:
```suggestion
- Remove event size check from the message receiver ([cafc722…](https://github.com/edgexfoundry/edgex-go/commit/cafc7220272db75275058deb04c0a161acd7b921))
```text

BREAKING CHANGE: Remove event size check from the message reciver, because we always check the event in the message publisher

```
```